### PR TITLE
7.x

### PR DESCRIPTION
**Pull request by daniel-dgi**

### CODE REVIEW

File: ucla_pdf.module [hunk 1]
```diff
@@ -131,41 +122,34 @@ function ucla_pdf_access($object = NULL, $dsid = NULL) {
...
+  
+  // Get the translation text from the datastrem
   module_load_include('inc', 'ucla_pdf', 'ucla_pdf');
   $out = ucla_pdf_format_datastream_text($object, $dsid);
-  $out .= "<br/><a href='download_translation'>Download as txt</a>";
+
+  // Add a link to donwload the text using Islandora's RESTful api.
+  global $base_url;
+  $download_url = $base_url . '/islandora/object/' . $object->id . '/datastream/' . $dsid . '/download';
+  $out .= "<br/><a class='ucla-pdf-link' href='{$download_url}'>Download as txt</a>";
```
Comment (jonathangreen) [inline ucla_pdf.module]: You should use this:
http://api.drupal.org/api/drupal/includes!common.inc/function/url/7

It is safer then concatenation, especially when dealing with translated sites.

**Comment by jonathangreen** [comment]
Can we also wrap the output on the tabs in pre tags, so that we retain formatting?

File: ucla_pdf.module [hunk 2]
```diff
@@ -131,41 +122,34 @@ function ucla_pdf_access($object = NULL, $dsid = NULL) {
...
+  // Get the translation text from the datastrem
   module_load_include('inc', 'ucla_pdf', 'ucla_pdf');
   $out = ucla_pdf_format_datastream_text($object, $dsid);
-  $out .= "<br/><a href='download_translation'>Download as txt</a>";
+
+  // Add a link to download the text using Islandora's RESTful api.
+  global $base_url;
+  $download_url = $base_url . '/islandora/object/' . $object->id . '/datastream/' . $dsid . '/download';
+  $link_text = t('Download as txt');
+  $out .= "<br /><pre>" . l($link_text, $download_url, array('attributes' => array('class' => array('ucla-pdf-link')))) . "</pre>";
```
Comment (jonathangreen) [inline ucla_pdf.module]: Can you pull out the pre tag around the link here?

Comment (daniel-dgi) [inline ucla_pdf.module]: I pushed a changeset to remove the pre tags.

**Pull request merged**